### DISCUSSION TITLE
vm_memory_high_watermark.relative is not aware of cgroups limits of the container on Kubernetes

### DISCUSSION THREAD

**Discussion by luarx** [rabbitmq/rabbitmq-server#10975] [Other] [answered]
### Describe the bug

Rabbitmq does not consider the container memory limit and it is using the total host memory

### Reproduction steps

I run the container in a Kubernetes node with 16Gi
The container has configured a memory limit: 1000Mi

- Use rabbitmq:3 image (https://hub.docker.com/layers/library/rabbitmq/3/images/sha256-0707df9f526c8ff49b4484c5ef4c8799e86fe198428d65817dccbac6de74609e)
- Configure vm_memory_high_watermark.relative = 0.7

### Expected behavior

I expect to see this watermark log:
_Memory high watermark setting: 0.7 of available memory, computed to: 700 mi_

Instead I see:
_Memory high watermark setting: 0.7 of available memory, computed to: 11.4611 gb_

**Comment by michaelklishin** [comment, marked as answer]
This is not a bug in RabbitMQ, and has been discussed many times before.

`vm_memory_high_watermark.relative` is not aware of anything Kubernetes-related. Its behavior completely depends on whether the runtime can determine the cgroups-capped value, and there are many factors that affect that, including the version of Kubernetes, the image OS, whether cgroups v2 are enabled, and so on.

Use [`vm_memory_high_watermark.absolute`](https://github.com/rabbitmq/rabbitmq-server/blob/v3.13.x/deps/rabbit/docs/rabbitmq.conf.example#L386-L402), which starting with 3.13.x even supports the same information unit suffixes as Kubernetes.

The list of Kubernetes-specific memory footprint and monitoring issues [does not end there](https://www.rabbitmq.com/docs/memory-use#page-cache), in particular for clusters where streams and super streams are used.

**Reply by luarx** [reply]
Thanks for that useful information! It helps me a lot 🙌

**Comment by michaelklishin** [comment]
[This doc guide](https://www.rabbitmq.com/docs/memory) is where `vm_memory_high_watermark.absolute` is covered.

I'll add a Kubernetes-specific note to both because this question keeps coming up.

**Reply by luarx** [reply]
Thanks you 🙌
